src/lib/ndpi_main.c

```diff
@@ -3030,6 +3030,15 @@ u_int16_t ndpi_guess_protocol_id(struct ndpi_detection_module_struct *ndpi_str,
 	    if (NDPI_ENTROPY_ENCRYPTED_OR_RANDOM(flow->entropy) != 0) {
 	      ndpi_set_risk(ndpi_str, flow, NDPI_SUSPICIOUS_ENTROPY);
 	    }
+
+	    struct ndpi_icmphdr * const icmphdr = (struct ndpi_icmphdr *)packet->payload;
+	    u_int16_t hdr_chksm = icmphdr->checksum;
+	    icmphdr->checksum = 0;
```

Comment:
The packet buffer will be modified, but the old state is restored later on.
It is required to calculate the correct checksum. I've decided against copying the whole packet buffer in to a new buffer before modification for the sake of performance.

Reply:
Not very fond of changing (even temporarily) the real packet, but I do understand the performance goal. Let's try this way anyway

Reply:
Solved those two issues by modifying the checksum calculation. Killing two birds with one stone. There is now need anymore for any buffer modification and the name of the function should make more sense now as the calculation is now much more different to other ones.